src/app/story/directives/player.component.spec.ts

```diff
@@ -64,6 +68,18 @@ describe('PlayerComponent', () => {
     expect(comp.episodeGuid).toEqual('episode1');
   });
 
+  cit('uses available enclosure url for unpublished', (fix, el, comp) => {
+    let dist = series.mock('prx:distribution', {kind: 'podcast', url: 'http://some-where'});
+    dist.mock('http://some-where', {publishedUrl: 'http://published-url', enclosurePrefix: 'http://prefix/'});
+    let storyDists = story.mockItems('prx:distributions', [{kind: 'episode', url: 'http://some-where/episode'}]);
+    storyDists[0].mock('http://some-where/episode', {guid: 'episode1', _links: { enclosure: { href: 'http://prefix/some-where/enclosure.mp3'}}});
+    comp.fromFeeder(new StoryModel(series, story), new DistributionModel(series, dist));
```

Comment:
Looks like maybe you had to call these methods directly rather than let the component do the thing, but I'm satisfied with this test. It might have worked to call ngDoCheck as well. I've had to do that with ngOnChanges because the component doesn't seem to go through the on change lifecycle hooks with tests.

Reply:
I truly have no idea if this is best, copied the style of the other tests for this part of the code.